go/vt/vtgate/tx_conn.go

```diff
@@ -150,3 +223,24 @@ func (txc *TxConn) multiGo(shardSessions []*vtgatepb.Session_ShardSession, actio
 	// simple aggregation to consolidate all errors.
 	return allErrors.Error()
 }
+
+// multiGoTargets executes the action for all targets in parallel and returns a consolildated error.
+// Flow is identical to multiGoSessions.
+func (txc *TxConn) multiGoTargets(targets []*querypb.Target, action func(*querypb.Target) error) error {
+	if len(targets) == 1 {
+		return action(targets[0])
+	}
+	allErrors := new(concurrency.AllErrorRecorder)
+	var wg sync.WaitGroup
+	for _, t := range targets {
+		wg.Add(1)
+		go func(t *querypb.Target) {
+			defer wg.Done()
+			if err := action(t); err != nil {
+				allErrors.RecordError(err)
+			}
+		}(t)
+	}
+	wg.Wait()
+	return allErrors.Error()
```

Comment:
In the previous (almost identical function), I debated this same issue and chose to anonymize the error. But I also see why it may be better to preserve the metadata.
For now, it's not very material because we don't retry anything during these flows.